### ISSUE TITLE
How can I get the scrolling props inside another component?

### ISSUE THREAD

**Issue by arturmiglio** [du5rte/react-skroll#10]
I want to setup some visual indicators (slider progress - the infamous bullets indicating which slide is active). 

This is my code: 
```
import React from 'react'
import Layout from '../components/layout'
import { FormattedMessage } from 'react-intl'
import { Scroller } from 'react-skroll'
import { css } from 'glamor'
import Button from '@material-ui/core/Button'

import { withIntl } from '../i18n'

const slides = [
  {
    index: 1,
    img: slide_1_img,
    color: '#3291C6',
  },
  ...
]

...

class IndexPage extends React.Component {
  render() {
    return (
      <Layout>
        <main css={{ position: 'fixed', height: '100%', width: '100%' }}>
          <Scroller
            scrollRef={ref => this.scroll = ref}
            autoScroll={true}
            autoFrame={true}
          >
            {slides.map(({ index, img, color }) => (
              <section
                key={index}
                id={`slide-${index}`}
                css={{
                  height: '100%',
                  width: '100%',
                  ...
                }}
              >
                {...}
              </section>
            ))}
          </Scroller>
          <SlideIndicators scrollProps={...} /> // How to get the props here???
        </main>
      </Layout>
    )
  }
}

export default withIntl(IndexPage)
```

I already tried to replicate de CodePen demo, but it seems that the library has been updated and there is no longer a <ScrollProvider>.

Any help would be very appreciated! Congrats for the library is super cool!

**Comment by arturmiglio** [comment]
It seems that ScrollProvider was injecting props on the class... But now, there is no ScrollProvider on the source code.

If I do a console.log on the render inside the scroller I can get the values, but they are not updated at each render, because it is a ref I suppose.

**Comment by du5rte** [comment]
You were on the right tracks but the outdated demos probably confused you. 

Using the [Reference Pattern](https://github.com/du5rte/react-skroll/blob/Documentation/README.md#reference-pattern) this should be all you need to do.

```js
<SlideIndicators scrollProps={this.scroll} />
```

I'm working on a much needed new documentation #11 feel free to contribute or even just comment or parts that are confusing.

**Comment by du5rte** [comment]
closes #9
